### ISSUE TITLE
Feature Request: ZFS Free/Used/Cap in Dashboard

### ISSUE THREAD

**Issue by g-mike** [webmin/webmin#1200]
Hi,

I'm using Webmin for a few years now together with ZFS.

Since the start of the project, my partition and used space grew a lot, and **I'd love to have option to change commands used to calculate free/used space together with capacity in the dashboard (Authentic Theme).**

Disk and network file systems are showing ZFS stats correctly, and the pool that concerns me is mounted as the root file system that I can easily query from the shell.

Oh, and the Kudos to the authors and supporters for a great job! :)

**Comment by chris001** [comment]
ZFS support would be a great addition, it's drastically superior to the default linux filesystems!

**Comment by jcameron** [comment]
The current dashboard should show ZFS free space already, but only for mounted filesystems - space not assigned to any filesystem is not included.

**Comment by g-mike** [comment]
Is there a way to select wich pool stats are displayed? Right now total cap for my pool is off by factor of 10 (7.2TB vs 180TB). `zfs list pool` shows stats correctly.

**Comment by jcameron** [comment]
What does `zfs list pool` output on your system exactly?

**Comment by g-mike** [comment]
Webmin displays:
```2.23 TiB used / 188.43 TiB free / 190.66 TiB total```

Whereas `zfs list rpool`
```NAME    USED  AVAIL  REFER  MOUNTPOINT
rpool  2.74T  4.28T    96K  /```

**Comment by abclution** [comment]
Hi guys,

Have you guys seen heard of zfsmanager webmin module?

https://github.com/jonmatifa/zfsmanager

Been using it for... uh about 5 years now I think?
It is fantastic and works great, really full featured..very solid.

There is a small incompatibility with the Authentic theme (can't remember exactly where the problem is, if I recall its during snapshot deletion?) so using it for certain actions needs to switch themes, as well as there are a few initial setup glitches with work around in the issues area, but after it works 100%.

It would really be a FINE addition to the other filesystem tools that webmin already has.

The original author, jonmatifa has been MIA for a while and there have been a few contributions by others as well.

@iliaross  @jcameron 


![image](https://user-images.githubusercontent.com/10130237/94987688-cab0ea00-0570-11eb-90eb-65921b16f53f.png)

**Comment by jcameron** [comment]
No, I haven't heard of that module! Does it work on Linux and/or Solaris?

ps - Nice meme

**Comment by abclution** [comment]
Linux 100%. (A few minor issues on installation, off the top of my head, the module sometimes doesn't show up or doesn't show up the right place, needs finagling in the webmin module assignment area. Minor things like that, I have a notebook somewhere with the probs / solutions.)
Reports of it working in BSD (and some patches from contributors regarding a few minor bsd incompatibilities IIRC from the Github issues, and Solaris I don't know but as far as I know, it uses and parses the standard zfs (zfs xxx, zpool xxx) tools output, as well as uses the same tools for its actions. So if ZFS runs on the systems, and Webmin runs on the systems, I bet it works.

**Comment by abclution** [comment]
Authentic Theme incompatibilities issue - https://github.com/jonmatifa/zfsmanager/issues/13

This issue occurs when generating a list of snapshots to delete, or listing them in general. Screen output gets wonky. 

ZFS Pools Missing from listing - https://github.com/jonmatifa/zfsmanager/issues/14  , https://github.com/jonmatifa/zfsmanager/issues/12
Needs minor config editing as output of zfs tools changed and zfsmanager needs more clues how to parse the output.

Missing link in menu - https://github.com/jonmatifa/zfsmanager/issues/11



Other than those 3 issues, in the last 4-5 years, I have had zero issues with it. I create/delete pools, datasets, snapshots, adjust zfs dataset properties etc etc. Its super well done.

I can't find the issue regarding the BSD patch but if I recall it was a minor gui issue (I could be mistaken of course.)

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
